pkg/sypgp/sypgp.go

```diff
@@ -376,144 +479,136 @@ func GetPassphrase(message string, retries int) (string, error) {
-	if err = StorePrivKey(entity); err != nil {
-		return
-	}
-	if err = StorePubKey(entity); err != nil {
-		return
-	}
+	fmt.Printf("done\n")
 
 	// Ask to push the new key to the keystore
-	pushKeyQ, err := AskQuestion("Would you like to push it to the keystore? [Y,n] : ")
-	if err != nil {
-		return
-	}
+	ans, err := askYNQuestion("y", "Would you like to push it to the keystore? [Y,n] ")
+	switch {
+	case err != nil:
+		fmt.Fprintf(os.Stderr, "Not pushing newly created key to keystore: %s\n", err)
 
-	if pushKeyQ == "" || pushKeyQ == "y" || pushKeyQ == "Y" {
+	case ans == "y":
 		err = PushPubkey(entity, keyServiceURI, authToken)
 		if err != nil {
-			return
+			fmt.Printf("Failed to push newly created key to keystore: %s\n", err)
+		} else {
+			fmt.Printf("Key successfully pushed to: %s\n", keyServiceURI)
 		}
-		fmt.Printf("Key successfully pushed to: %v\n", keyServiceURI)
+
+	default:
+		fmt.Printf("NOT pushing newly created key to: %s\n", keyServiceURI)
 	}
-	fmt.Printf("Done.\n")
 
-	return
+	return entity, nil
 }
 
 // DecryptKey decrypts a private key provided a pass phrase
 func DecryptKey(k *openpgp.Entity, message string) error {
+	if !k.PrivateKey.Encrypted {
+		return errNotEncrypted
+	}
+
 	if message == "" {
 		message = "Enter key passphrase : "
 	}
-	if k.PrivateKey.Encrypted {
-		pass, err := AskQuestionNoEcho(message)
-		if err != nil {
-			return err
-		}
 
-		if err := k.PrivateKey.Decrypt([]byte(pass)); err != nil {
-			return err
-		}
+	pass, err := AskQuestionNoEcho(message)
+	if err != nil {
+		return err
 	}
-	return nil
+
+	return k.PrivateKey.Decrypt([]byte(pass))
 }
 
 // EncryptKey encrypts a private key using a pass phrase
-func EncryptKey(k *openpgp.Entity, pass string) (err error) {
+func EncryptKey(k *openpgp.Entity, pass string) error {
 	if k.PrivateKey.Encrypted {
 		return fmt.Errorf("key already encrypted")
 	}
-	err = k.PrivateKey.Encrypt([]byte(pass))
-	return
+	return k.PrivateKey.Encrypt([]byte(pass))
 }
 
 // SelectPubKey prints a public key list to user and returns the choice
 func SelectPubKey(el openpgp.EntityList) (*openpgp.Entity, error) {
-	PrintPubKeyring()
+	printEntities(os.Stdout, el)
 
-	index, err := AskQuestion("Enter # of public key to use : ")
-	if err != nil {
-		return nil, err
-	}
-	if index == "" {
-		return nil, fmt.Errorf("invalid key choice")
-	}
-	i, err := strconv.ParseUint(index, 10, 32)
+	n, err := askNumberInRange(0, len(el)-1, "Enter # of public key to use : ")
 	if err != nil {
 		return nil, err
 	}
 
-	if i < 0 || i > uint64(len(el))-1 {
-		return nil, fmt.Errorf("invalid key choice")
-	}
-
-	return el[i], nil
+	return el[n], nil
 }
 
 // SelectPrivKey prints a secret key list to user and returns the choice
 func SelectPrivKey(el openpgp.EntityList) (*openpgp.Entity, error) {
-	PrintPrivKeyring()
+	printEntities(os.Stdout, el)
 
-	index, err := AskQuestion("Enter # of signing key to use : ")
+	n, err := askNumberInRange(0, len(el)-1, "Enter # of signing key to use : ")
```

Comment:
```suggestion
	n, err := askNumberInRange(0, len(el)-1, "Enter # of private key to use : ")
```

Because when you export a key.

Reply:
Done, thanks.